pytroll_collectors/tests/test_segments.py

```diff
@@ -651,6 +670,38 @@ def test_check_and_add_existing_files(self, glob, logging):
             assert "Only 'file' messages are supported." in logs
         assert logging.disable.call_count == 2
 
+    @patch("pytroll_collectors.segments.logging")
+    @patch("fsspec.filesystem")
+    def test_check_and_add_existing_s3_files(self, filesystem, logging):
+        """Test that existing matching files are added to the slot in S3 storage."""
+        # There's no 's3://' in the glob result of fsspec.filesystem("s3").glob() call
+        existing_files = [
+            "bucket-name/H-000-MSG3__-MSG3________-VIS006___-000007___-201611281100-__",
+            "bucket-name/H-000-MSG3__-MSG3________-VIS006___-000008___-201611281100-__",
+            # A file that should not match
+            "bucket-name/H-000-MSG4__-MSG4________-VIS006___-000008___-201611281100-__"]
+        glob = MagicMock()
+        glob.return_value = existing_files
+        s3 = MagicMock()
+        s3.glob = glob
+        filesystem.return_value = s3
+        mda = self.mda_msg0deg_s3.copy()
+        fake_message = FakeMessage(mda)
+        message = Message(fake_message, self.msg0deg._patterns['msg'])
+        self.msg0deg._create_slot(message)
+        slot_str = str(mda["start_time"])
+        slot = self.msg0deg.slots[slot_str]
+        self.msg0deg._config['check_existing_files_after_start'] = True
```

Comment:
This looks very similar to the previous test, can it be refactored to avoid duplication?

Reply:
Restructured in b0b769843210f35bc22d952d4fbf9d15de99d2bb